livre/core/types.h

```diff
@@ -112,25 +133,53 @@ typedef boost::shared_ptr< DashProcessor > DashProcessorPtr;
 typedef boost::shared_ptr< ProcessorInput > ProcessorInputPtr;
 typedef boost::shared_ptr< ProcessorOutput > ProcessorOutputPtr;
 typedef boost::shared_ptr< Renderer > RendererPtr;
-typedef boost::shared_ptr< const View > ConstViewPtr;
 typedef boost::shared_ptr< View > ViewPtr;
 typedef boost::shared_ptr< GLContext > GLContextPtr;
 typedef boost::shared_ptr< const GLContext > ConstGLContextPtr;
 typedef boost::shared_ptr< GLWidget > GLWidgetPtr;
-typedef boost::shared_ptr< VolumeInformation > VolumeInformationPtr;
-typedef boost::shared_ptr< const VolumeInformation > ConstVolumeInformationPtr;
 typedef boost::shared_ptr< TextureState > TextureStatePtr;
 typedef boost::shared_ptr< const TextureState > ConstTextureStatePtr;
 typedef boost::shared_ptr< VolumeDataSource > VolumeDataSourcePtr;
 typedef boost::shared_ptr< const VolumeDataSource > ConstVolumeDataSourcePtr;
 typedef boost::shared_ptr< EventHandler > EventHandlerPtr;
 typedef boost::shared_ptr< EventHandlerFactory > EventHandlerFactoryPtr;
-typedef boost::shared_ptr< EventMapper > EventMapperPtr;
 typedef boost::shared_ptr< MemoryUnit > MemoryUnitPtr;
 typedef boost::shared_ptr< const MemoryUnit > ConstMemoryUnitPtr;
-
+typedef boost::shared_ptr< PortData > PortDataPtr;
+typedef boost::shared_ptr< Filter > FilterPtr;
+typedef boost::shared_ptr< const Filter > ConstFilterPtr;
+typedef boost::shared_ptr< PortData > PortDataPtr;
+typedef boost::shared_ptr< const PortData > ConstPortDataPtr;
+typedef boost::shared_ptr< Pipeline > PipelinePtr;
+typedef boost::shared_ptr< const Pipeline > ConstPipelinePtr;
+typedef boost::shared_ptr< PipeFilter > PipeFilterPtr;
+typedef boost::shared_ptr< const PipeFilter > ConstPipeFilterPtr;
+typedef boost::shared_ptr< Workers > WorkersPtr;
+typedef boost::shared_ptr< Executor > ExecutorPtr;
+typedef boost::shared_ptr< InputPort > InputPortPtr;
+typedef boost::shared_ptr< OutputPort > OutputPortPtr;
+typedef boost::shared_ptr< const InputPort > ConstInputPortPtr;
+typedef boost::shared_ptr< const OutputPort > ConstOutputPortPtr;
+typedef boost::shared_ptr< LODEvaluator > LODEvaluatorPtr;
+typedef boost::shared_ptr< const LODEvaluator > ConstLODEvaluatorPtr;
+typedef boost::shared_ptr< Cache > CachePtr;
+typedef boost::shared_ptr< const Cache > ConstCachePtr;
+typedef boost::shared_ptr< Executable > ExecutablePtr;
+typedef boost::shared_ptr< const Executable > ConstExecutablePtr;
+typedef boost::shared_ptr< CacheObject > CacheObjectPtr;
+typedef boost::shared_ptr< const CacheObject > ConstCacheObjectPtr;
+typedef boost::shared_ptr< Future > FuturePtr;
+typedef boost::shared_ptr< const Future > ConstFuturePtr;
+typedef boost::shared_ptr< Promise > PromisePtr;
+typedef boost::shared_ptr< const Promise > ConstPromisePtr;
 typedef boost::shared_ptr< CacheObject > CacheObjectPtr;
 typedef boost::shared_ptr< const CacheObject > ConstCacheObjectPtr;
+typedef boost::shared_ptr< GLSLShaders > GLSLShadersPtr;
```

Comment:
Since you're going to eventually move this into a separate library, is there any reason for not using std::shared_ptr from the beginning?